zipline/cli/commands/upload.py

```diff
@@ -0,0 +1,195 @@
+    ] = None,
+    override_name: Annotated[
+        Optional[str],
+        Option(
+            "--name",
+            "-n",
+            help="Specify the name to give this file in Zipline. This overrides the --format option, if provided.",
+        ),
+    ] = None,
+    original_name: Annotated[
+        Optional[bool],
+        Option(
+            "--original-name/--generated-name",
+            "-o/-g",
+            help="Specify whether the original name of the file should be preserved when downloading the file from Zipline.",
+        ),
+    ] = False,
+    folder: Annotated[
+        Optional[str],
+        Option(
+            help="Specify what folder the file should be added to after it is uploaded.",
+        ),
+    ] = None,
+    override_extension: Annotated[
+        Optional[str],
+        Option(
+            "--extension",
+            "-e",
+            help="Specify what file extension the file should be uploaded with. This should only really be used if the library fails to determine what kind of file your file is.",
+        ),
+    ] = None,
+    override_domain: Annotated[
+        Optional[str],
+        Option(
+            "--domain",
+            "-d",
+            help="Specify what domain should be used instead of the Zipline instance's core domain.",
+        ),
+    ] = None,
+) -> None:
+    """Upload a file to a remote Zipline instance."""
+    with Progress(
+        SpinnerColumn(),
+        TextColumn("[progress.description]{task.description}"),
+        transient=True,
+    ) as progress:
+        task = progress.add_task(description="Preparing client...", total=None)
+        client = Client(server_url, token)
+
+        progress.update(task, description="Reading file...", total=None)
+        file_data = FileData(
+            data=file,
+            filename=override_name or file.name,
+        )
+
+        progress.update(task, description="Uploading file...", total=None)
+        try:
+            uploaded_file = await client.upload_file(
+                payload=file_data,
+                compression_percent=compression_percent,
+                expiry=expiry,
+                format=format,
+                password=password,
+                max_views=max_views,
+                override_name=override_name,
+                original_name=original_name,
+                folder=folder,
+                override_extension=override_extension,
+                override_domain=override_domain,
+                text_only=True,
+            )
+        except NonHttpUrlClientError as e:
+            echo(f"Invalid URL provided: '{server_url}'", err=True)
+            await client.close()
+            raise Exit(1) from e
+        except (NotAuthenticated, Forbidden) as e:
+            echo("Authentication failure! Are you using a valid token?", err=True)
+            await client.close()
+            raise Exit(77) from e
+        except BadRequest as e:
+            echo("Bad request!", err=True)
```

Comment:
Will this expose any information that is useful to the user? Such as the error message for what actually went wrong?